python_modules/libraries/dagster-databricks/dagster_databricks/components/databricks_workspace/component.py

```diff
@@ -0,0 +1,136 @@
+import asyncio
+import re
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import Annotated, Any, Optional
+
+from dagster import AssetExecutionContext, AssetKey, AssetSpec, Definitions, Resolvable, multi_asset
+from dagster._annotations import beta
+from dagster.components import Resolver
+from dagster.components.component.state_backed_component import StateBackedComponent
+from dagster.components.utils.defs_state import (
+    DefsStateConfig,
+    DefsStateConfigArgs,
+    ResolvedDefsStateConfig,
+)
+from dagster_shared.serdes.serdes import deserialize_value, serialize_value
+from databricks.sdk import WorkspaceClient
+
+from dagster_databricks.components.databricks_asset_bundle.configs import Job
+from dagster_databricks.components.databricks_workspace.fetcher import (
+    fetch_databricks_workspace_data,
+)
+from dagster_databricks.components.databricks_workspace.schema import (
+    AssetSpecConfig,
+    DatabricksFilterConfig,
+    DatabricksWorkspaceConfig,
+    resolve_databricks_filter,
+)
+
+
+def _snake_case(name: str) -> str:
+    name = re.sub(r"[^a-zA-Z0-9]+", "_", str(name))
+    name = re.sub(r"([a-z0-9])([A-Z])", r"\1_\2", name)
+    return name.lower().strip("_")
+
+
+@beta
+@dataclass
+class DatabricksWorkspaceComponent(StateBackedComponent, Resolvable):
+    """Component that fetches Databricks workspace jobs and exposes them as assets."""
+
+    workspace: Annotated[
+        DatabricksWorkspaceConfig,
+        Resolver.default(description="Databricks workspace connection info"),
+    ]
+
+    databricks_filter: Annotated[
+        Optional[DatabricksFilterConfig],
+        Resolver.default(description="Filter which Databricks jobs to include"),
+    ] = None
```

Comment:
this should be of type `Optional[DatabricksFilter]` and then it should have `Resolver(resolve_databricks_filter, model_field_type=Optional[DatabricksFilterModel], description=...)`

that way we automatically resolve the filter when loading this class from yaml